### DISCUSSION TITLE
How can I make fields readonly or hidden in MudExObjectEdit?

### DISCUSSION THREAD

**Discussion by Quotic** [fgilde/MudBlazor.Extensions#54] [Q&A] [answered]
Hi,

I don't find a easy answer to this. It is possible to hide or make readonly some fields of an object? Or need I to create a displayClass.
i.E. I have a Database object to edit. so I don't want that the user change the ID field or the CreatedOn value.
Can you explain me, how to do this?

Thanks

**Comment by fgilde** [comment, marked as answer]
You can do this in the meta config either by bypassing Configuration to the control `<MudExObjectEditForm MetaConfiguration="@Configure" Value="@Model"></MudExObjectEditForm>`  or with just implementing somewhere in your project the `IObjectMetaConfiguration<TModel>` interface.

Then inside the meta config you can access any property or properties you want and set any option, render config etc.

That means for your case something like this.
```c#
    private void Configure(ObjectEditMeta<Model> meta)
    {
        meta.Property(m => m.Id).AsReadOnly()
        meta.Property(m => m.CreatedOn).AsReadOnly()
    }
```

**Reply by Quotic** [reply]
Thanks Florian, it is working! I have done it this way and grouped the properties .WithGroup("Database") is it possible to collapse this group automatically? And completly hiding of properties isn't possible?

**Comment by fgilde** [comment]
sure it is

```c#
    private void Configure(ObjectEditMeta<Model> meta)
    {
        meta.Property(m => m.Id).Ignore();  // Ignor property Id
        meta.Property(c => c.SomeObjectWithProperties).Children.Recursive(om => om.Children).Ignore(); // Ignore a property with all containing properties
    }
```

You can also have more complex conditions if you want to just ignore when something is true, and also with the same conditions you can set all attributes and properties as you want 

for example this is something I've used in one of my apps

```c#
            meta.Properties()
                .AsReadOnlyIf(m => !CanEdit(m))
                .WithAttributesIf(m => !CanEdit(m),
                    new KeyValuePair<string, object>(nameof(MudTextField<string>.Class), "mud-ex-keep-input-label-visible"),
                    new KeyValuePair<string, object>(nameof(MudTextField<string>.Variant), Variant.Text),
                    new KeyValuePair<string, object>(nameof(MudTextField<string>.DisableUnderLine), true),
                    new KeyValuePair<string, object>(nameof(MudTextField<string>.HelperText), string.Empty)
            );
            meta.Property(m => m.PatientBirthday).IgnoreIf(m => !CanEdit(m));

```

And then in my case this is how my can edit method looks like

```c#
    private bool CanEdit(CaseDto caseDto)
    {
        if (ForceReadOnly)
            return false;
        if (_currentUser == null)
            return false;
        if (CanEditAsAdmin)
            return true;
        if (!_currentUser.HasPolicy(Permissions.Cases.Edit) && !IsNewCase)
            return false;
        if (!_currentUser.HasPolicy(Permissions.Cases.Create) && IsNewCase)
            return false;
        return IsNewCase || (caseDto.Status != CaseStatusDto.Closed && IsOwnCase);
    }

```

**Reply by Quotic** [reply]
Thanks for your fast support!
